tests/test_models/test_losses.py

```diff
@@ -87,3 +89,95 @@ def test_accuracy():
+    self = BboxGIoU2D()
+    bboxes1, nb_bbox = _construct_bbox()
+    bboxes2, _ = _construct_bbox(nb_bbox)
+    bboxes1 = torch.cat((bboxes1, torch.rand((nb_bbox, 1))), 1)
+    bboxes2 = torch.cat((bboxes2, torch.rand((nb_bbox, 1))), 1)
+    gious = self(bboxes1, bboxes2, True)
+    assert gious.size() == (nb_bbox, ), gious.size()
+    assert torch.all(gious >= -1) and torch.all(gious <= 1)
+
+    # is_aligned is True, bboxes1.size(-2) == 0
+    bboxes1 = torch.empty((0, 4))
+    bboxes2 = torch.empty((0, 4))
+    gious = self(bboxes1, bboxes2, True)
+    assert gious.size() == (0, ), gious.size()
+    assert torch.all(gious == torch.empty((0, )))
+    assert torch.all(gious >= -1) and torch.all(gious <= 1)
+
+    # is_aligned is True, and bboxes.ndims > 2
+    bboxes1, nb_bbox = _construct_bbox()
+    bboxes2, _ = _construct_bbox(nb_bbox)
+    bboxes1 = bboxes1.unsqueeze(0).repeat(2, 1, 1)
+    # test assertion when batch dim is not the same
+    with pytest.raises(AssertionError):
+        self(bboxes1, bboxes2.unsqueeze(0).repeat(3, 1, 1), True)
+    bboxes2 = bboxes2.unsqueeze(0).repeat(2, 1, 1)
+    gious = self(bboxes1, bboxes2, True)
+    assert torch.all(gious >= -1) and torch.all(gious <= 1)
+    assert gious.size() == (2, nb_bbox)
+    bboxes1 = bboxes1.unsqueeze(0).repeat(2, 1, 1, 1)
+    bboxes2 = bboxes2.unsqueeze(0).repeat(2, 1, 1, 1)
+    gious = self(bboxes1, bboxes2, True)
+    assert torch.all(gious >= -1) and torch.all(gious <= 1)
+    assert gious.size() == (2, 2, nb_bbox)
+
+    # is_aligned is False
+    bboxes1, nb_bbox1 = _construct_bbox()
+    bboxes2, nb_bbox2 = _construct_bbox()
+    gious = self(bboxes1, bboxes2)
+    assert torch.all(gious >= -1) and torch.all(gious <= 1)
+    assert gious.size() == (nb_bbox1, nb_bbox2)
+
+    # is_aligned is False, and bboxes.ndims > 2
+    bboxes1 = bboxes1.unsqueeze(0).repeat(2, 1, 1)
+    bboxes2 = bboxes2.unsqueeze(0).repeat(2, 1, 1)
+    gious = self(bboxes1, bboxes2)
+    assert torch.all(gious >= -1) and torch.all(gious <= 1)
+    assert gious.size() == (2, nb_bbox1, nb_bbox2)
+    bboxes1 = bboxes1.unsqueeze(0)
+    bboxes2 = bboxes2.unsqueeze(0)
+    gious = self(bboxes1, bboxes2)
+    assert torch.all(gious >= -1) and torch.all(gious <= 1)
+    assert gious.size() == (1, 2, nb_bbox1, nb_bbox2)
+
+    # is_aligned is False, bboxes1.size(-2) == 0
+    gious = self(torch.empty(1, 2, 0, 4), bboxes2)
+    assert torch.all(gious == torch.empty(1, 2, 0, bboxes2.size(-2)))
+    assert torch.all(gious >= -1) and torch.all(gious <= 1)
+
+    # test allclose between bbox_gious and the original official
+    # implementation.
+    bboxes1 = torch.FloatTensor([
+        [0, 0, 10, 10],
+        [10, 10, 20, 20],
+        [32, 32, 38, 42],
+    ])
+    bboxes2 = torch.FloatTensor([
+        [0, 0, 10, 20],
+        [0, 10, 10, 19],
+        [10, 10, 20, 20],
+    ])
+    gious = bbox_gious(bboxes1, bboxes2, is_aligned=True, eps=eps)
+    gious = gious.numpy().round(4)
```

Comment:
round(4) here is unnecessary?

Reply:
It's necessary so as to get small enough (e.g. 1e-7) absolute error, since the gious_gt is got with four decimal precision.